File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int len=heights.size();
        int max_area=(len-1)*min(heights[0],heights[len-1]);
        int left=0;
        int right=len-1;
        int curr_area;
        while(left<right)
        {
            curr_area=(right-left)*min(heights[left],heights[right]);
            if(curr_area>max_area)
            {
                max_area=curr_area;
            }
            if(heights[left]<=heights[right])
            {
                left++;
            }
            else
            {
                right--;
            }
            
        }
        return max_area;
    }
};

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        def depth(node):
            if node==None:
                return [0,0]
            t = depth(node.right)
            a = t[0]
            d1 = t[1]
            t = depth(node.left)
            b = t[0]
            d2 = t[1]
            diameter = max([d1,d2, a+b+1])
            return [(max(a,b) + 1),diameter]
        depth,diameter = depth(root)
        return diameter - 1


        